server/migrations/2020-06-30-135809_remove_mat_views/up.sql

```diff
@@ -0,0 +1,576 @@
+
+
+-- Drop the mviews
+drop view post_mview;
+drop materialized view user_mview;
+drop view community_mview;
+drop materialized view private_message_mview;
+drop view user_mention_mview;
+drop view reply_view;
+drop view comment_mview;
+drop materialized view post_aggregates_mview;
+drop materialized view community_aggregates_mview;
+drop materialized view comment_aggregates_mview;
+
+-- User
+create table user_fast as select * from user_view;
+alter table user_fast add column fast_id serial primary key;
+
+create index idx_user_fast_id on user_fast (id);
+
+drop trigger refresh_user on user_;
+
+create trigger refresh_user
+after insert or update or delete
+on user_
+for each row
```

Comment:
The procedures if you look, has a switch on the action type, from the examples for trigger procedures in the postgres docs: https://www.postgresql.org/docs/9.5/plpgsql-trigger.html

 > For ex. if a user changes their password

Luckily it just updates that specific row in the `user_fast` table anyway. We do have an auto-updating timestamp `updated` column.

Reply:
The action based `ifs` are different from a `when` defined https://www.postgresql.org/docs/9.1/sql-createtrigger.html - My suggestion is to apply a `when` to updates specifically so that you avoid the case where the ORM applies a new updated timestamp but the aggregates don't need refreshed.  Ex:
```
create trigger refresh_user_updates on user_
after update for each row when (old.column != new.column)
```
and replace column with what specific columns should trigger refreshed because regardless of the end result being 1 row updated, you probably don't want refreshes firing off errant or irrelevant changes.